strategy/sampling/sampling_rule.go

```diff
@@ -31,19 +31,19 @@ type Properties struct {
 // AppliesTo returns true if the sampling rule matches against given parameters. False Otherwise.
 // Assumes lock is already held, if required.
 func (p *Properties) AppliesTo(host, path, method string) bool {
-	return (host == "" || pattern.WildcardMatchCaseInsensitive(p.Host, host)) &&
-		(path == "" || pattern.WildcardMatchCaseInsensitive(p.URLPath, path)) &&
-		(method == "" || pattern.WildcardMatchCaseInsensitive(p.HTTPMethod, method))
+	return pattern.WildcardMatchCaseInsensitive(p.Host, host) &&
+		pattern.WildcardMatchCaseInsensitive(p.URLPath, path) &&
+		pattern.WildcardMatchCaseInsensitive(p.HTTPMethod, method)
 }
 
 // AppliesTo returns true if the sampling rule matches against given sampling request. False Otherwise.
 // Assumes lock is already held, if required.
 func (r *CentralizedRule) AppliesTo(request *Request) bool {
```

Comment:
Changing this to exclude absent request attributes would lead to behavior inconsistent with other SDKs, for example our [node SDK](https://github.com/aws/aws-xray-sdk-node/blob/master/packages/core/lib/middleware/sampling/sampling_rule.js#L38-L42) and [Python SDK](https://github.com/aws/aws-xray-sdk-python/blob/master/aws_xray_sdk/core/sampling/sampling_rule.py#L45-L49). I understand the ask to change this, but changing the behavior like this would be a fairly consequential, arguably breaking, change.

@kevioke I believe the change you made to the `beginSegment` API should be sufficient to resolve #242 since the `serviceName` will no longer be absent. Our team will have to discuss this change which may delay this PR quite a bit. If you'd like to remove this change and only include the `beginSegment` change that could expedite this getting released.

Reply:
@willarmiros thanks for the reply! I had a hunch this empty request logic was present for a reason, just didn't know quite why. In any case, I like your proposal of just including the changes to `beginSegment` to speed this PR along, and I believe you're correct that it should work for my use cases since now I can specify a rule with a ServiceName.

I'll update the PR later today or tomorrow.